db/migrate/20180329141702_add_plan_and_segment_tables.rb

```diff
@@ -0,0 +1,27 @@
+class AddPlanAndSegmentTables < ActiveRecord::Migration[5.1]
+  def change
+    create_table :plans do |t|
+      t.string :name
+      t.string :type
+      t.timestamps
+    end
+    create_table :timeframes do |t|
+      t.belongs_to :plan, index: true
+      t.date :start
+      t.date :end
+      t.timestamps
+    end
+    create_table :segments do |t|
+      t.belongs_to :timeframe, index: true
+      t.belongs_to :road, index: true
+      t.boolean :is_whole_road
```

Comment:
This is a philosophical comment more than something that actually needs to be changed: 

As programmers, we are constantly checking to see if the state of something is true. Truthiness can often be found in most dynamic languages in just about every type as long as there is a value assigned. This being the case, truthiness should be used as an active state, not the default. Since our interface has the default state being a whole road, the default for `:is_whole_road` is `true`. It is more sound to have the column named `:is_segment` so we can adhere to our default state when checking for truthiness.

Reply:
Additionally (this is something that probably should be included if we are being defensive and thorough), the nature of booleans is that there are only two possible values. If Postgres allows `NULL` in boolean columns, `typeof null !== 'boolean'`. Since your default state assumes `true` based on the reason in my comment above, `null == false` which would will trigger conditionals to not be triggered that we may want to.  

**tl;dr** Always set default values for boolean columns based upon the default state that the column represents.

Reply:
I was attempting to match the way that we asked the question in the UI where the TRUE position indicates that the user is talking about the whole road. But I see your point and will change this.